### PR TITLE
Add observing run from observation plan

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR creates an observing run from an observation plan.

### CODE REVIEW

File: skyportal/tests/frontend/test_scanning_page.py [hunk 1]
```diff
@@ -840,7 +840,7 @@ def test_user_without_save_access_cannot_save(
         f'//button[@name="initialSaveCandidateButton{public_candidate.id}"]'
     )
     driver.scroll_to_element_and_click(save_button)
-    driver.wait_for_xpath("//*[contains(.,'Insufficient permissions')]")
+    # driver.wait_for_xpath("//*[contains(.,'Insufficient permissions')]")
```
Comment (Theodlz) [inline skyportal/tests/frontend/test_scanning_page.py:843]: Should this just be deleted before merging if its not used anymore ?

File: static/js/components/ObservationPlanRequestLists.jsx [hunk 2]
```diff
@@ -202,6 +202,14 @@ const ObservationPlanRequestLists = ({ gcnEvent }) => {
     setIsDeletingTreasureMap(null);
   };
 
+  const [isCreatingObservingRun, setIsCreatingObservingRun] = useState(null);
+  const handleCreateObservingRun = async (id) => {
+    console.log("id", id);
```
Comment (Theodlz) [inline static/js/components/ObservationPlanRequestLists.jsx:207]: remove console.log

File: skyportal/handlers/api/followup_request.py [hunk 3]
```diff
@@ -54,6 +55,60 @@
...
+        Database session for this transaction
+    """
+
+    user = session.query(User).get(user_id)
+
+    try:
+        assignment = ClassicalAssignment(**AssignmentSchema.load(data=data))
+    except ValidationError as e:
+        raise ValidationError(
+            'Error parsing followup request: ' f'"{e.normalized_messages()}"'
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/followup_request.py:74]: We don't have a self available when running like this is the issue.

**Comment by Theodlz** [comment]
More general comment on this review. A lot of the logic found in the post methods of handlers are now extracted from the handlers to be refactored as methods that can be used by the handler but also reused in other handlers, which is super useful (thanks for doing that !). In most of them, I don't see any try catch statement, I'm not sure they would be necessary, but could it be useful to add some try catch either around where you call the method or in the method itself, to catch some of the errors and display error messages in the logs ?

File: skyportal/handlers/api/followup_request.py [hunk 4]
```diff
@@ -54,6 +55,60 @@
...
+            ClassicalAssignment.obj_id == assignment.obj_id,
+            ClassicalAssignment.run_id == run_id,
+        )
+        .first()
+    )
+
+    if predecessor is not None:
+        raise ValueError('Object is already assigned to this run.')
+
+    assignment = ClassicalAssignment(**data)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/followup_request.py:93]: https://github.com/skyportal/skyportal/blob/main/skyportal/handlers/api/followup_request.py#L163
I think this is about setting priority correctly.

File: static/js/components/ObservationPlanRequestLists.jsx (hunk 2 above)
Comment (mcoughlin) [reply static/js/components/ObservationPlanRequestLists.jsx:207]: yes

File: skyportal/tests/frontend/test_scanning_page.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/tests/frontend/test_scanning_page.py]: yeah

**Comment by mcoughlin** [comment]
@Theodlz good point!

**Review by Theodlz — APPROVED** [review]
No further comments. Would have been nice for the results of the simsurvey to open in another tab to not close skyportal tabs that can take a while to load some times, but its not essential.

File: skyportal/handlers/api/followup_request.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Theodlz) [reply skyportal/handlers/api/followup_request.py:74]: Oh ! I am just talking about the error message itself (like the string) not the way the error is raised!

**Review by thomasculino — APPROVED** [review]
Same remark for the results of the simsurvey, otherwise it looks good

**Comment by mcoughlin** [comment]
@Theodlz @thomasculino Yes I think down the link we might give the simsurvey stuff it's own page (or something) to allow for different injection numbers, light curves, etc. Can revisit maybe with the analysis framework stuff.

**Pull request merged**